Leaderboard cron

zushar opened the pull request:
for dtaft onle

zushar committed:
in summary add the bass for cron job for leaderboard db

zushar committed:
a

barel-mishal commented on src/leaderboard/leaderboard.service.ts:
```diff
@@ -2,12 +2,82 @@ import { AnalyticsDto } from '@/common/dto/leaderboard';
+
+  @Cron(CronExpression.EVERY_DAY_AT_MIDNIGHT)
+  async handleCron() {
+    console.log('Running fetch and store members cron job');
```
That’s great @zushar! Very good implementation for the data insertion.

Before merging, I believe we need to make a few changes:

	1.	In the handleCron function, we should fetch new data only on Sundays. Fetching and processing this data daily is unnecessary. Please modify the CronExpression.EVERY_DAY_AT_MIDNIGHT to run every Sunday at 8:00 AM.
	2.	We need an additional function. Each time we fetch data, we retrieve all the records. While this may not be the most efficient method, it is the approach provided by GitHub. Consequently, this will accumulate a significant amount of data. Please add a function to delete all existing records before storing the new data as you currently do.

Let me know if you need any further adjustments!

zushar replied:
done

zushar committed:
chore: Refactor leaderboard service

The leaderboard service has been refactored to improve code readability and maintainability. The changes include:
- Renamed the `memberModel` variable to `ContributorModel` for better clarity.
- Added a new method `deleteAllMembers()` to delete all members from the database.
- Updated the cron expression to run the cron job at 8:00 AM every Sunday.
- Replaced console log message from "Running fetch and store members cron job" to "Running fetch and store Contributors cron job".
- Updated the function name `saveFetchedFromGithubMemberToDb` to `saveFetchedFromGithubContributorToDb`.
- Updated the query in the `getLeaderBoardData` method to use the `ContributorModel` instead of `memberModel`.

These changes aim to improve the overall functionality and maintainability of the leaderboard service.

barel-mishal commented on src/leaderboard/leaderboard.service.spec.ts:
```diff
@@ -1,11 +1,40 @@
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [
+        TestDbModule,
+        // MongooseModule.forRoot('mongodb://127.0.0.1:27017/test'),
```
Does the comment necessary?

barel-mishal commented on src/leaderboard/leaderboard.service.ts:
```diff
@@ -2,12 +2,84 @@ import { AnalyticsDto } from '@/common/dto/leaderboard';
+    await this.handleCron();
+  }
+
+  @Cron('0 8 * * 0') // Cron expression for 8:00 AM every Sunday
```
Excellent
